weave-js/src/common/components/WandbLoader.tsx

```diff
@@ -86,3 +88,48 @@ export const TrackedWandbLoader = ({
 
   return <WandbLoader {...props} />;
 };
+
+export const TrackedWaveLoader = ({
+  captureException,
+  name,
+  profilingCb,
+  samplingRate = 0.1,
+  track,
+  onComplete,
+  onStart,
+  size,
+}: TrackedWandbLoaderProps & {
+  size: 'small' | 'huge';
+}) => {
+  useLifecycleProfiling(
+    name,
+    (data: ProfileData) => {
+      try {
+        // log the lifecycle for each loader to segment
+        const additionalData = profilingCb ? profilingCb() : {};
+        const trackedData = {
+          componentId: data.id,
+          duration: data.duration,
+          ...additionalData,
+        };
+        if (onComplete) {
+          onComplete(name, trackedData);
+        }
+        const randomNum = Number(Math.random().toString().slice(-2)); // take the last two digits off a random number
```

Comment:
**nitpick:** looks like this is just copied from above but not a fan of the number->string->number acrobatics here. something like `Math.floor(Math.random() * 100)` feels more elegant and is likely faster (though i'm sure the cost is extremely negligible 😛)

Reply:
story of my life 😆